client/bin/client.js

```diff
@@ -1,18 +1,114 @@
 #!/usr/bin/env node
 
-import { join, dirname } from "path";
-import { fileURLToPath } from "url";
-import handler from "serve-handler";
+import fs from "fs";
 import http from "http";
+import https from "https";
+import { dirname, join } from "path";
+import handler from "serve-handler";
+import { fileURLToPath } from "url";
 
 const __dirname = dirname(fileURLToPath(import.meta.url));
 const distPath = join(__dirname, "../dist");
 
+// Function to determine the MCP server URL
+const getMcpServerUrl = () => {
+  if (process.env.MCP_PROXY_FULL_ADDRESS) {
+    return process.env.MCP_PROXY_FULL_ADDRESS;
+  }
+
+  // Use current host with custom port if specified
+  const port = process.env.SERVER_PORT || "6277";
+  // Default to http://localhost:port
+  return `http://localhost:${port}`;
```

Comment:
```suggestion
  return `http://127.0.0.1:${port}`;
```

Does that suggested change look correct?